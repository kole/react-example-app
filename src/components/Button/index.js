import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Color from 'color';

const ButtonContainer = styled.div`
    button {
        background: ${props => props.buttonColor};
        border: 1px solid #777;
        color: ${props => props.textColor};
        cursor: pointer;
        font-size: 1.2em;
        outline: none;
        padding: 1.1em 1.4em;
        text-transform: uppercase;

        :hover {
            background: ${props => Color(props.buttonColor).lighten(0.5).string()};
        }
    }
`;

const Button = (props) => {
    const { text, clickHandler, buttonColor, textColor } = props;
    return (
        <ButtonContainer buttonColor={buttonColor} textColor={textColor}>
            <button onClick={clickHandler}>{text}</button>
        </ButtonContainer>
    )
};

Button.propTypes = {
    clickHandler: PropTypes.func,
    text: PropTypes.string.isRequired,
    textColor: PropTypes.string,
    buttonColor: PropTypes.string
};

Button.defaultProps = {
    buttonColor: '#004b79',
    textColor: '#fff'
}

export default Button;
