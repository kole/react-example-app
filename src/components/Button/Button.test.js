/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index.js';

describe('Button', () => {
    test('Snapshot', () => {
        const component = renderer
            .create(
                <Button
                    buttonColor="red"
                    clickHandler={() => {}}
                    text="Test Button text"
                    textColor="#fff"
                />
            )
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
