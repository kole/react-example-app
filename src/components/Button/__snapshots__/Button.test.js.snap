// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<div
  className="sc-bdVaJa hmzYzR"
>
  <button
    onClick={[Function]}
  >
    Test Button text
  </button>
</div>
`;
